receiver/googlecloudspannerreceiver/internal/metadata/metricsmetadatatometrics_test.go

```diff
@@ -0,0 +1,195 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadata
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/model/pdata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/datasource"
+)
+
+func TestMetricsMetadata_ToMetrics(t *testing.T) {
+	testCases := map[string]struct {
+		metricsDataType pdata.MetricDataType
+	}{
+		"Gauge": {pdata.MetricDataTypeGauge},
+		"Sum":   {pdata.MetricDataTypeSum},
+	}
+
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			testMetricsMetadataToMetrics(t, testCase.metricsDataType)
+		})
+	}
+}
+
+func testMetricsMetadataToMetrics(t *testing.T, metricDataType pdata.MetricDataType) {
+	timestamp := time.Now().UTC()
+	labelValues := allPossibleLabelValues()
+	metricValues := allPossibleMetricValues(metricDataType)
+	databaseID := databaseID()
+	metadata := MetricsMetadata{MetricNamePrefix: metricNamePrefix}
+
+	metrics := metadata.toMetrics(databaseID, timestamp, labelValues, metricValues)
+
+	assert.Equal(t, len(metricValues), len(metrics))
+
+	for i, metric := range metrics {
+		assert.Equal(t, 1, metric.DataPointCount())
+		assert.Equal(t, 1, metric.MetricCount())
+		assert.Equal(t, 1, metric.ResourceMetrics().Len())
+		assert.Equal(t, 1, metric.ResourceMetrics().At(0).InstrumentationLibraryMetrics().Len())
+		assert.Equal(t, 1, metric.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0).Metrics().Len())
+
+		ilMetric := metric.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0).Metrics().At(0)
+
+		assert.Equal(t, metadata.MetricNamePrefix+metricValues[i].Name(), ilMetric.Name())
+		assert.Equal(t, metricValues[i].Unit(), ilMetric.Unit())
+		assert.Equal(t, metricValues[i].DataType().MetricDataType(), ilMetric.DataType())
+
+		var dataPoint pdata.NumberDataPoint
+
+		if metricDataType == pdata.MetricDataTypeGauge {
+			assert.NotNil(t, ilMetric.Gauge())
+			assert.Equal(t, 1, ilMetric.Gauge().DataPoints().Len())
+			dataPoint = ilMetric.Gauge().DataPoints().At(0)
+		} else {
+			assert.NotNil(t, ilMetric.Sum())
+			assert.Equal(t, pdata.MetricAggregationTemporalityDelta, ilMetric.Sum().AggregationTemporality())
+			assert.True(t, ilMetric.Sum().IsMonotonic())
+			assert.Equal(t, 1, ilMetric.Sum().DataPoints().Len())
+			dataPoint = ilMetric.Sum().DataPoints().At(0)
+		}
+
+		assertMetricValue(t, metricValues[i], dataPoint)
+
+		assert.Equal(t, pdata.NewTimestampFromTime(timestamp), dataPoint.Timestamp())
+		assert.Equal(t, 3+len(labelValues), dataPoint.Attributes().Len())
```

Comment:
What are the extra 3 label values? project, instance and database?

Reply:
Yes, you are correct. Each Spanner DB is identified by projectID, instanceID and database. I added them to metric labels to identify to which DB metric relates.